### PR TITLE
Add initialization of node and network object directly from application - Closes #4710

### LINKED ISSUE #4710: Move chain module to the application level
### Description

The chain module should live inside of the controller.
Also, it would be nice to change the name to `node` instead of `chain`.

### Motivation

Related to #3358 , chain module also is not pluggable module, and it requires the code injection (such as custom transactions)

### Acceptance Criteria

Folder chain module is moved and all the code are rewired.

### Additional Information

- [x] Adjust the file structure to move modules/chain to contoller/node
- [x] Update the directory structure for better file organization https://github.com/LiskHQ/lisk-sdk/issues/4751
- [x] Add initialization of node object directly from application
- [x] Add initialization of network object directly from application
- [x] Register all node events and actions to application level channel
- [x] Update references of chain event emitting 
- [ ] Update configuration structure and move chain inside app namespace 
- [ ] Update the configuration fixtures to match new structure
- [ ] Write migration to migrate namespace "chain" in the migrations table. 
- [x] Create issues on core to update config for networks and also write config migration https://github.com/LiskHQ/lisk-core/issues/211

### PR DESCRIPTION
**Pull request by nazarhussain**
### What was the problem?

The `network` and `node` was initialized in controller, although semantically they are part of the application level now. 

### How did I solve it?

Moved all initialization logic of `network` and `node` to application level. 

### How to manually test it?

Run all tests. 

### Review checklist

- [ ] The PR resolves #4710
- [ ] All new code is covered with unit tests
- [ ] Relevant integration / functional tests are added
- [ ] Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
- [ ] Documentation has been added/updated

### CODE REVIEW

File: framework/src/controller/controller.js [hunk 1]
```diff
@@ -58,29 +49,19 @@ class Controller {
...
 		await this._setupBus();
-		await this._loadMigrations({ ...migrations, app: controllerMigrations() });
-		await this._initialiseNetwork(networkConfig);
+		await this._loadMigrations({ ...migrations });
 		await this._loadModules(modules, moduleOptions);
 
 		this.logger.debug(this.bus.getEvents(), 'Bus listening to events');
 		this.logger.debug(this.bus.getActions(), 'Bus ready for actions');
-
-		this.channel.publish('app:ready');
```
Comment (shuse2) [inline framework/src/controller/controller.js]: where is this `app:ready` published now?

Comment (nazarhussain) [reply framework/src/controller/controller.js]: From the application. after loading up the everything including controller.

File: framework/src/application/node/node.js [hunk 2]
```diff
@@ -61,31 +61,6 @@ module.exports = class Node {
 	}
 
 	async bootstrap() {
```
Comment (ishantiw) [inline framework/src/application/node/node.js]: Can we have `initialiseNode` to be consistent with the network

Comment (nazarhussain) [reply framework/src/application/node/node.js]: Normally initialization meant to be constructor called. Object is initialized when called the constructor. So I kept it bootstrap explicitly with this in mind.

Comment (ishantiw) [reply framework/src/application/node/node.js]: then we can rename the network's `initialiseNetwork` -> `bootstrap`

**Pull request merged**